src/shared/components/plots/StackedBarPlot.tsx

```diff
@@ -0,0 +1,630 @@
+import * as React from "react";
+import {Observer, observer} from "mobx-react";
+import {IStringAxisData} from "../../../pages/resultsView/plots/PlotsTabUtils";
+import {computed, observable} from "mobx";
+import {bind} from "bind-decorator";
+import CBIOPORTAL_VICTORY_THEME, {axisTickLabelStyles} from "../../theme/cBioPoralTheme";
+import {getTextWidth} from "../../lib/wrapText";
+import autobind from "autobind-decorator";
+import _ from "lodash";
+import {VictoryAxis, VictoryBar, VictoryChart, VictoryLabel, VictoryStack, VictoryLegend} from "victory";
+import Timer = NodeJS.Timer;
+import {tickFormatNumeral} from "./TickUtils";
+import {makeUniqueColorGetter} from "./PlotUtils";
+import {stringListToIndexSet} from "../../lib/StringUtils";
+import ScatterPlotTooltip from "./ScatterPlotTooltip";
+
+export interface IStackedBarPlotProps {
+    svgId?:string;
+    domainPadding?:number;
+    horzData:IStringAxisData["data"];
+    vertData:IStringAxisData["data"];
+    categoryToColor?:{[cat:string]:string};
+    barWidth:number;
+    chartBase:number;
+    horizontalBars?:boolean;
+    horzCategoryOrder?:string[];
+    vertCategoryOrder?:string[];
+    axisLabelX?: string;
+    axisLabelY?: string;
+    legendLocationWidthThreshold?: number;
+}
+
+const RIGHT_GUTTER = 120; // room for legend
+const NUM_AXIS_TICKS = 8;
+const PLOT_DATA_PADDING_PIXELS = 100;
+const CATEGORY_LABEL_HORZ_ANGLE = 50;
+const DEFAULT_LEFT_PADDING = 25;
+const DEFAULT_BOTTOM_PADDING = 10;
+const LEGEND_ITEMS_PER_ROW = 4;
+const BOTTOM_LEGEND_PADDING = 15;
+const RIGHT_PADDING_FOR_LONG_LABELS = 50;
+const COUNT_AXIS_LABEL = "# samples";
+
+@observer
+export default class StackedBarPlot extends React.Component<IStackedBarPlotProps, {}> {
+    @observable.ref tooltipModel:any|null = null;
+    @observable pointHovered:boolean = false;
+    private mouseEvents:any = this.makeMouseEvents();
+
+    @observable.ref private container:HTMLDivElement;
+
+    @bind
+    private containerRef(container:HTMLDivElement) {
+        this.container = container;
+    }
+
+    @computed get getColor() {
+        const uniqueColorGetter = makeUniqueColorGetter(_.values(this.props.categoryToColor));
+        const categoryToColor:{[category:string]:string} = {};
+        _.forEach(this.props.categoryToColor, (color, category)=>{
+            categoryToColor[category.toLowerCase()] = color;
+        });
+        return function(category:string) {
```

Comment:
i think you could make this function a method of the class the references a computed value categoryToColor.   it's maybe a little clearer?

Reply:
i dont understand your sentence, maybe some typos?